packages/flutter_tools/test/commands.shard/hermetic/attach_test.dart

```diff
@@ -1302,6 +1306,13 @@ class FakeIOSDevice extends Fake implements IOSDevice {
   @override
   final IOSDeviceConnectionInterface interfaceType;
 
+  @override
+  DeviceConnectionInterface get connectionInterface {
```

Comment:
I know you added this enum in a previous refactor PR #121359, but is there any reason to maintain both `DeviceConnectionInterface` and `IOSDeviceConnectionInterface`?  Can we swap the iOS device to use `DeviceConnectionInterface` instead, with the `none` either removed, or added to `DeviceConnectionInterface`?

Reply:
Good point, on further inspection, I think we can safely get rid of IOSDeviceConnectionInterface. The `none` case is for simulators, which we consider `attached` and aren't even included in `IOSDevice`. The "unknown future interfaces" gets categorized as `usb` currently, which will get changed to `attached`. 

https://github.com/flutter/flutter/blob/fcf11343ae9e291189ba59dc27c3d3e7def17ff9/packages/flutter_tools/lib/src/macos/xcdevice.dart#L360-L371